ngraph/test/frontend/paddlepaddle/op/pool2d.cpp

```diff
@@ -0,0 +1,482 @@
+                                                                    {72.50, 74.50 }},
+                                                                    {{80.50, 82.50 },
+                                                                    {88.50, 90.50 }}}}})
+                                .get_vector());
+
+        return res;
+    }
+
+    static avgPool2DTestParam avgPool_test2() {
+        avgPool2DTestParam res;
+        res.m_frontEndName = PDPD;
+        res.m_modelsPath =   PATH_TO_MODELS;
+        res.m_modelName =    "avgPool_test2";
+
+        // data (2, 3, 4, 4) input tensor
+        res.inputs.emplace_back(shared_input_NCHW);
+
+        // (1, 1, 4, 4)
+        res.expected_outputs.emplace_back(test::NDArray<float, 4>({{{{{7.5 }},
+                                                                    {{23.5 }},
+                                                                    {{39.5 }}},
+                                                                    {{{55.5 }},
+                                                                    {{71.5 }},
+                                                                    {{87.5 }}}}})
+                                .get_vector());
+
+        return res;
+    }
+
+    static avgPool2DTestParam avgPool_test3() {
+        avgPool2DTestParam res;
+        res.m_frontEndName = PDPD;
+        res.m_modelsPath =   PATH_TO_MODELS;
+        res.m_modelName =    "avgPool_test3";
+
+        // data (2, 3, 4, 4) input tensor
+        res.inputs.emplace_back(shared_input_NCHW);
+
+        // (2, 3, 2, 2)
+        res.expected_outputs.emplace_back(test::NDArray<float, 4>({{{{{2.50, 4.50 },
+                                                                    {10.50, 12.50 }},
+                                                                    {{18.50, 20.50 },
+                                                                    {26.50, 28.50 }},
+                                                                    {{34.50, 36.50 },
+                                                                    {42.50, 44.50 }}},
+                                                                    {{{50.50, 52.50 },
+                                                                    {58.50, 60.50 }},
+                                                                    {{66.50, 68.50 },
+                                                                    {74.50, 76.50 }},
+                                                                    {{82.50, 84.50 },
+                                                                    {90.50, 92.50 }}}}})
+                                .get_vector());
+
+        return res;
+    }
+
+    static avgPool2DTestParam avgPool_test4() {
+        avgPool2DTestParam res;
+        res.m_frontEndName = PDPD;
+        res.m_modelsPath =   PATH_TO_MODELS;
+        res.m_modelName =    "avgPool_test4";
```

Comment:
Please also generate and upload actual model files to ngraph/test/files/paddlepaddle/models folder. C++ unit tests are not intended to run python scripts to generate models. In this case such models have to be in repository, like 'prototxt' files for ONNX

Reply:
Ok, I will upload all the test models